agent/agentbootstrap/bootstrap.go

```diff
@@ -425,12 +425,19 @@ func initBootstrapMachine(
 	if args.BootstrapMachineHardwareCharacteristics != nil {
 		hardware = *args.BootstrapMachineHardwareCharacteristics
 	}
+
 	hostSeries, err := series.HostSeries()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	spaceAddrs, err := args.BootstrapMachineAddresses.ToSpaceAddresses(st)
```

Comment:
These addresses come from `instance.Addresses()`, which coming from the provider, are `ProviderAddresses`.

At this point, spaces have been loaded from the provider, so conversion to `SpaceAddresses` via lookup will succeed.